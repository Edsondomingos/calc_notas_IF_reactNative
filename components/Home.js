import * as React from 'react';
import { useState } from 'react';
import { Text, View, StyleSheet, TouchableHighlight } from 'react-native';
import Styles from '../assets/Styles'

export default function Home(){
  return (
    <View style={Styles.container}>
      <TouchableHighlight
        style={Styles.opcao}
        onPress={() => console}
      >
        <Text style={Styles.textoOpcao}>Médio/Técnico</Text>
      </TouchableHighlight>

      <TouchableHighlight
        style={Styles.opcao}
        onPress={() => console}
      >
        <Text style={Styles.textoOpcao}>Superior</Text>
      </TouchableHighlight>
      
      <TouchableHighlight
        style={Styles.opcao}
        onPress={() => console}
      >
        <Text style={Styles.textoOpcao}>Sobre</Text>
      </TouchableHighlight>
    </View>
  )
}