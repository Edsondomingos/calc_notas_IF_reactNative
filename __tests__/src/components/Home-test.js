import renderer from 'react-test-renderer'
import { render, fireEvent } from "@testing-library/react-native";
import Home from '../../../components/Home';
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe('Teste de snapshot', () => {
    it('Snapshot da tela Home', () => {
        const tela = renderer.create(<Home />).toJSON()
        expect(tela).toMatchSnapshot()
    })    
})

describe('Teste de componentes tela Home', () => {
    it('Botão medio/tecnico', () => {
        const {getByText, getByTestId } = render(<Home />)

        const titulo = getByText('Calculadora de notas')
        expect(titulo).toBeTruthy()
        // expect(titulo).toBeDefined()

        // const bt = getByTestId('btnTelaCalcMedio')
        // fireEvent.press(bt)

        // expect(getByText('Calculadora Medio')).toBeTruthy()

        
    })
    it('Botão Superior', () => {
        const { getByText, getByTestId} = render(<Home />);
    
        const botao = getByTestId("btnTelaCalcSuperior")
        fireEvent.press(botao)

        expect(getByText('Tela Dobro')).toBeTruthy()
                
    })
})

// describe("Teste de Navegação", () => {
//     it('Botão medio/tecnico', () => {
    // const {getByText, getByTestId } = render(<Home />)
    // expect(titulo).toBeDefined()

    // const bt = getByTestId('btnTelaCalcMedio')
    // fireEvent.press(bt)

    // expect(getByText('Calculadora Medio')).toBeTruthy()

    
    // })
    // it('Botão Superior', () => {
    //     const { getByText, getByTestId} = render(<Home />);

    //     const botao = getByTestId("btnTelaCalcSuperior")
    //     fireEvent.press(botao)

    //     expect(getByText('Tela Dobro')).toBeTruthy()
                
    // })
// })