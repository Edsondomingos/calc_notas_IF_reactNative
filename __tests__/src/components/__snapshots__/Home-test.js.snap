// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot Snapshot da tela Home 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "marginHorizontal": "auto",
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "Cursive",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginVertical": 20,
        "textAlign": "center",
      }
    }
  >
    Calculadora de notas
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f05",
        "borderBottomEndRadius": 20,
        "borderTopEndRadius": 20,
        "borderWidth": 1,
        "height": 70,
        "justifyContent": "center",
        "margin": "auto",
        "marginVertical": 30,
        "padding": 10,
        "width": 200,
      }
    }
    testID="btnTelaCalcMedio"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Cursive",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Médio➗Técnico
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f05",
        "borderBottomEndRadius": 20,
        "borderTopEndRadius": 20,
        "borderWidth": 1,
        "height": 70,
        "justifyContent": "center",
        "margin": "auto",
        "marginVertical": 30,
        "padding": 10,
        "width": 200,
      }
    }
    testID="btnTelaCalcSuperior"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Cursive",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      ➕Superior
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f05",
        "borderBottomEndRadius": 20,
        "borderTopEndRadius": 20,
        "borderWidth": 1,
        "height": 70,
        "justifyContent": "center",
        "margin": "auto",
        "marginVertical": 30,
        "padding": 10,
        "width": 200,
      }
    }
    testID="btnTelaSobre"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Cursive",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      ❕Sobre
    </Text>
  </View>
</View>
`;
